File: apps/android/app/src/main/java/com/ramoieeeufjf/appRamo/security/AccessPolicy.kt
package com.ramoieeeufjf.appRamo.security

object AccessPolicy {
    const val GLOBAL_CHAPTER = "Todos"
    const val UNASSIGNED_CHAPTER = "Sem capítulo"

    private val privilegedRoles = setOf(
        "admin",
        "administrador",
        "diretoria",
        "presidente",
        "vice presidente",
        "vice-presidente",
        "tesoureiro",
        "webmaster"
    )

    fun visibleChapters(chapterRoles: Map<String, String>): List<String> {
        return (chapterRoles.keys.filter { it.isNotBlank() } + GLOBAL_CHAPTER)
            .distinct()
            .take(10)
    }

    fun canManageContent(chapterRoles: Map<String, String>): Boolean {
        return chapterRoles.any { (chapter, role) ->
            chapter.equals("Diretoria", ignoreCase = true) || isPrivilegedRole(role)
        }
    }

    fun canControlRoom(chapterRoles: Map<String, String>): Boolean {
        return canManageContent(chapterRoles)
    }

    fun publicChapterLabels(chapterRoles: Map<String, String>): List<String> {
        return chapterRoles.keys
            .filter { it.isNotBlank() && it != GLOBAL_CHAPTER }
            .ifEmpty { listOf(UNASSIGNED_CHAPTER) }
    }

    private fun isPrivilegedRole(role: String): Boolean {
        val normalized = role.trim().lowercase()
        return normalized in privilegedRoles
    }
}

File: apps/ios/AppRamoIEEE/AccessPolicy.swift
import Foundation

enum AccessPolicy {
    static let globalChapter = "Todos"
    static let unassignedChapter = "Sem capítulo"

    private static let privilegedRoles: Set<String> = [
        "admin",
        "administrador",
        "diretoria",
        "presidente",
        "vice presidente",
        "vice-presidente",
        "tesoureiro",
        "webmaster"
    ]

    static func visibleChapters(_ chapterRoles: [String: String]) -> [String] {
        var chapters = chapterRoles.keys.filter { !$0.trimmingCharacters(in: .whitespacesAndNewlines).isEmpty }
        chapters.append(globalChapter)
        var unique: [String] = []
        for chapter in chapters where !unique.contains(chapter) {
            unique.append(chapter)
        }
        return Array(unique.prefix(10))
    }

    static func canManageContent(_ chapterRoles: [String: String]) -> Bool {
        chapterRoles.contains { chapter, role in
            chapter.caseInsensitiveCompare("Diretoria") == .orderedSame || isPrivilegedRole(role)
        }
    }

    static func canControlRoom(_ chapterRoles: [String: String]) -> Bool {
        canManageContent(chapterRoles)
    }

    static func publicChapterLabels(_ chapterRoles: [String: String]) -> [String] {
        let chapters = chapterRoles.keys
            .filter { !$0.trimmingCharacters(in: .whitespacesAndNewlines).isEmpty }
            .filter { $0 != globalChapter }
            .sorted()
        return chapters.isEmpty ? [unassignedChapter] : chapters
    }

    private static func isPrivilegedRole(_ role: String) -> Bool {
        privilegedRoles.contains(role.trimmingCharacters(in: .whitespacesAndNewlines).lowercased())
    }
}
